src/datasets/features.py

```diff
@@ -169,13 +171,13 @@ def _cast_to_python_objects(obj: Any) -> Tuple[Any, bool]:
         import jax.numpy as jnp
 
     if isinstance(obj, np.ndarray):
-        return obj.tolist(), True
+        return obj.tolist(), False
     elif config.TORCH_AVAILABLE and "torch" in sys.modules and isinstance(obj, torch.Tensor):
-        return obj.detach().cpu().numpy().tolist(), True
+        return obj.detach().cpu().numpy(), True
     elif config.TF_AVAILABLE and "tensorflow" in sys.modules and isinstance(obj, tf.Tensor):
-        return obj.numpy().tolist(), True
+        return obj.numpy(), True
     elif config.JAX_AVAILABLE and "jax" in sys.modules and isinstance(obj, jnp.ndarray):
-        return obj.tolist(), True
+        return jnp.asarray(obj), True
```

Comment:
`jnp.asarray` is to convert an array to a jnp.ndarray, you can use `np.array` instead:
```suggestion
        return np.array(obj), True
```

Reply:
Oh, I saw only `ndarray` return type and not `np`. My bad :/